### ISSUE TITLE
RL-0004: Add Authentication and Authorization using JWT

### ISSUE THREAD

**Issue by anoop-pandiri** [anoop-pandiri/rl#7]
**Existing Dependencies:**
spring-boot-starter
spring-boot-starter-web
spring-boot-starter-test

**Add Dependencies:**
spring-boot-starter-data-jpa
spring-boot-starter-validation
spring-boot-starter-security
h2db
jjwt-api
jjwt-impl
jjwt-jackson
lombok
dotenv-java
springdoc-openapi-starter-webmvc-ui

**.env file to store:**
DATABASE_URL
DATABASE_USERNAME
DATABASE_PASSWORD
JWT_SECRET_KEY

new endpoints: **"/register"**, **"/login"**

TODO: Authentication & Authorization

**Comment by anoop-pandiri** [comment]
The schema is still under review. Further details and decisions are needed on the following aspects
Currently the general rule is to use singular, all caps table names with no special characters, including underscores. This also applies to columns as well.
Whether to use OpenAPI spec and OpenAPI code generator for controllers.
A Minimalistic design documentation may be needed to fully capture and finalize the schema design, naming conventions, constraints, and API specifications
Issue will be On Hold until further notice.

This Issue is blocked by #8
